### PR TITLE
yamidecode: Don't render frame before decoding frame started.

### PR DESCRIPTION
**Pull request by zhongcong**
Signed-off-by: Zhong Cong <congx.zhong@intel.com>

### CODE REVIEW

**Comment by zhongcong** [comment]
updated.
in processOneFrame(), if renderOneFrame() return RENDER_NO_AVAILABLE_FRAME, just return, don't need to invoke renderDone() function.

File: tests/decodeoutput.cpp [hunk 1]
```diff
@@ -59,6 +59,9 @@ Decode_Status DecodeOutput::processOneFrame(bool drain)
 {
     Decode_Status status = renderOneFrame(drain);
 
+    if (status == RENDER_NO_AVAILABLE_FRAME)
```
Comment (xuguangxin) [inline tests/decodeoutput.cpp]: we only renderDone picture when status == RENDER_SUCCESS. Can we change code to this
...
if (status == RENDER_SUCCESS) {
      m_renderFrames++;
      m_decoder->renderDone....
}
return status;
}

**Comment by zhongcong** [comment]
updated.

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged as 8d64174072a3611c40bcfd0c63a6fd8457779cd6

**Pull request closed without merging**